boilerplate/update

```diff
@@ -14,41 +14,101 @@ if [ "$BOILERPLATE_SET_X" ]; then
     set -x
 fi
 
-DIR="$(realpath $(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd ))"
-
-# Were we re-exec'd?
-if [ "$1" ]; then
-  OUT=$1
-  # Let's make sure this was really a re-exec and not some other
-  # weirdness.
-  if ! [[ -d "$OUT" ]] || ! diff -q "$0" "$OUT/boilerplate/update"; then
-    echo "Something went wrong! I was invoked with '$OUT'!"
+# The directory in which this script lives is the CONVENTION_ROOT. Export
```

Comment:
Changes to this file should be (slightly) more scrutable if you look at one commit at a time.